src/types/profile.js

```diff
@@ -316,6 +316,42 @@ export type ExtensionTable = {|
   length: number,
 |};
 
+/**
+ * Visual progress describes the visual progression during page load. A sample is generated
+ * everytime the visual completeness of the webpage changes.
+ */
+export type VisualProgress = {|
+  // A percentage that describes the visual completeness of the webpage, ranging from 0% - 100%
+  percent: number,
+  // The time in milliseconds which the sample was taken.
+  timestamp: Milliseconds,
+|};
+
+/**
+ * Visual metrics are performance metrics that measure above-the-fold webpage visual performance,
+ * and more accurately describe how human end-users perceive the speed of webpage loading. These
+ * metrics serves as additional metrics to the typical page-level metrics such as onLoad. Visual
+ * metrics contains key metrics such as Speed Index, Perceptual Speed Index, and ContentfulSpeedIndex,
+ * along with their visual progression. These properties find their way into the gecko profile through running
+ * browsertime, which is a tool that lets you collect performance metrics of your website.
+ * Browsertime provides the option of generating a gecko profile, which then embeds these visual metrics
+ * into the geckoprofile. More information about browsertime can be found through https://github.com/mozilla/browsertime.
```

Comment:
This link is not providing me with information about what the individual values represent:

For instance: `PerceptualSpeedIndexProgress` isn't in the codebase.

I think it would be helpful to link to where this data gets generated. It's often nice to jump directly to the serializer to see what's going on.

Reply:
I've added links to where the data gets generated in the new update